### PR TITLE
Add JWT authentication

### PR DESCRIPTION
**Pull request by artemklishch**
BREAKING CHANGES:

*Added authentication token classes

Task: https://mate.academy/learn/java-spring-boot/java-spring-boot-security-jwt/linear?learnItemsFilter=All&section=practice&videoId=5197&theoryId=3217&testTaskSlug=java_add_jwt_support

### CODE REVIEW

File: src/main/java/org/example/intro/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class OpenApiConfig {
+
+  @Bean
+  public OpenAPI customOpenApi() {
+    return new OpenAPI()
+      .components(new Components().addSecuritySchemes("BearerAuth",
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/config/OpenApiConfig.java]: create constant for BearerAuth
check whitespaces (maybe I am wrong, but looks strange)

File: src/main/java/org/example/intro/config/SecurityConfig.java [hunk 2]
```diff
@@ -2,23 +2,29 @@
...
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
 @EnableWebSecurity
 @EnableMethodSecurity
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/config/SecurityConfig.java]: Do you need both of these annotations `@EnableWebSecurity and @EnableMethodSecurity`

File: src/main/java/org/example/intro/dto/user/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package org.example.intro.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotEmpty
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/dto/user/UserLoginRequestDto.java]: NotBlank for String

File: src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                    null,
+                    userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java]: create constant for "Bearer " and use it length() in `bearerToken.substring(7);`

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/intro/config/SecurityConfig.java (hunk 2 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/config/SecurityConfig.java]: ++, removed. it looks like it works without this annotation

File: src/main/java/org/example/intro/config/OpenApiConfig.java (hunk 1 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/config/OpenApiConfig.java]: ++, thank, updated

File: src/main/java/org/example/intro/dto/user/UserLoginRequestDto.java (hunk 3 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/dto/user/UserLoginRequestDto.java]: ++, thanks. updated

File: src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java (hunk 4 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java]: ++, thank you. updated.

File: src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java (hunk 5 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/sequrity/JwtAuthenticationFilter.java]: ++, thanks. updated

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**